packages/service-clients/end-to-end-tests/azure-client/src/test/tree.spec.ts

```diff
@@ -72,35 +72,28 @@ for (const testOpts of testMatrix) {
 			client = createAzureClient();
 		});
 
-		/**
-		 * Scenario: test when an Azure Client container is created,
-		 * it can set the initial objects to SharedTree and do basic SharedTree ops.
-		 *
-		 * Expected behavior: an error should not be thrown nor should a rejected promise
-		 * be returned.
-		 */
-		it("can create a container with SharedTree and do basic ops", async () => {
+		async function init(
+			summaryTree: (typeof ephemeralSummaryTrees)[keyof typeof ephemeralSummaryTrees],
+		): Promise<{ containerId: string; treeData: TreeView<typeof StringArray> }> {
 			let containerId: string;
-			let container1: IFluidContainer;
 			let treeData: TreeView<typeof StringArray>;
+
+			let container1: IFluidContainer;
 			if (isEphemeral) {
 				const containerResponse: AxiosResponse | undefined = await createContainerFromPayload(
-					ephemeralSummaryTrees.createContainerWithSharedTree,
+					summaryTree,
 					"test-user-id-1",
 					"test-user-name-1",
 				);
+
 				containerId = getContainerIdFromPayloadResponse(containerResponse);
 				({ container: container1 } = await client.getContainer(containerId, schema, "2"));
 
-				treeData = (container1.initialObjects.tree1 as ITree).viewWith(
-					treeConfiguration, // This is defined in schema.ts
-				);
+				treeData = (container1.initialObjects.tree1 as ITree).viewWith(treeConfiguration);
```

Comment:
container schema is suppose to have strong typing so we should not need that cast. Maybe move container1's definition inside the if, and let it infer the type?

Reply:
Good idea, I got rid of the underspecified `let` binding so we can have the stronger typing.